src/caching_key_wrapper.rs

```diff
@@ -141,14 +145,14 @@ impl<K> CachingKeyWrapper<K> {
         None
     }
 
-    async fn get_cached_decryption_key(&self, encrypted_key: &Vec<u8>) -> Option<Key<U16>> {
+    async fn get_cached_decryption_key(&self, encrypted_key: &[u8]) -> Option<Key<S>> {
         let mut decryption_cache = self.decryption_cache.lock().await;
 
         if let Some(cached_key) = decryption_cache.get(encrypted_key) {
             // Only return the cached key if the age is less than the max_age param.
             // I don't think this is strictly necessary, but it's what the JS AWS SDK does.
             if cached_key.created_at.elapsed() <= self.options.max_age {
-                return Some(cached_key.key);
+                return Some(cached_key.key.clone());
```

Comment:
Why is the clone required here?

Reply:
I could be wrong, but I think previously, key here impl Copy, so we didn't need clone. But with the generic keysize `S`, that's no longer the case, so had to be cloned.